arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -538,6 +537,23 @@ void fvm_lowered_cell_impl<Backend>::initialize(
     for (auto cell_idx: make_span(ncell)) {
         cell_gid_type gid = gids[cell_idx];
 
+        // Sanity check recipe
```

Comment:
This could be simplified slightly, and have the tests on the variables closer to where they are defined.
This will make opting out of specific checks easier if we decide to allow that later for performance reasons.
```C++
auto& cell = cells[cell_idx];
if (rec.num_source(gid) != cell.detectors().size()) {
    throw ...;
}
auto cell_targets = util::sum_by(cell.synapses(), [](auto& syn) {return syn.second.size();});
if (num_targets != rec.num_targets(gid)) {
    throw ...;
}
```

Reply:
Sure.